### PR TITLE
[WIP] (DO NOT MERGE UNTIL UNIT TESTS ARE WRITTEN) add fasta dataset loader

### PR DESCRIPTION
**Pull request by alat-rights**
# Pull Request Template

## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by alat-rights** [comment]
I'd like to add a truncated uniprot file to the S3 so that we can incorporate it in unit tests for this PR, since lack of sharding support means we cannot load all of uniprot yet.

Would that be ok?

**Comment by alat-rights** [comment]
A little concerned about whether FASTALoader will be able to handle the fact that Union[str, featurizer] is accepted by the loader class.

File: deepchem/molnet/load_function/fasta_datasets.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  def create_dataset(self) -> Dataset:
+    dataset_file = os.path.join(self.data_dir, "sequences.fasta")
+    if not os.path.exists(dataset_file):
+      dc.utils.data_utils.download_url(url=FASTA_URL, dest_dir=self.data_dir)
+    loader = dc.data.FASTALoader(featurizer=self.featurizer, legacy=False)
+    return loader.create_dataset(dataset_file, shard_size=8192)
+
+
+def load_FASTA(
```
Comment (rbharath) [inline deepchem/molnet/load_function/fasta_datasets.py:24]: MoleculeNet loaders are intended for particular datasets (like Uniprot/Swissprot etc). We already have a FASTALoader class for loading general FASTA files

**Review by rbharath — COMMENTED** [review]
I think this PR isn't needed perhaps? Explanation in the comment below

**Comment by rbharath** [comment]
Closing this old PR for cleanup

**Pull request closed without merging**